### PR TITLE
API Protocol and dynamic NS-entry handling

### PR DESCRIPTION
**Pull request by bfbones**
Hello,

as wished my changes for your new revision. I know my implementiation is a little bit dirty, but works fine. It is a little bit tricky to handle the entries of jtables dynamically, so I workarounded it a little bit.
This changes enable the application to handle a unknown amount of NS-entries from config and from template without any problem. Nothing hardcoded (just two NS-entries) any more.
I hope you like it and if you have improvements feel free to mention them.

Thanks

### CODE REVIEW

**Comment by bfbones** [comment]
Is there any problem with these changes?
If not, you can merge them :)

File: includes/config.inc.php-dist [hunk 1]
```diff
@@ -5,6 +5,7 @@ $apipass = '';          # The PowerDNS API-user password or the PowerDNS-API key
 $apiip   = '';          # The IP of the PowerDNS API
 $apiport = '8081';      # The port of the PowerDNS API
 $apisid  = 'localhost'; # PowerDNS's :server_id
+$apiprotocol = 'https://'; # protocol used to connect with trailing slashs
```
Comment (JonasGroeger) [inline includes/config.inc.php-dist]: Calling it `$apiprotocol = 'https';` makes this setting a little bit more configuration-like.

File: zones.php [hunk 2]
```diff
@@ -11,14 +11,14 @@
 }
 
 function api_request($path, $opts = null, $type = null) {
-    global $apisid, $apiuser, $apipass, $apiip, $apiport, $authmethod;
+    global $apisid, $apiuser, $apipass, $apiip, $apiport, $authmethod, $apiprotocol;
 
-    $url = "http://$apiip:$apiport${path}";
+    $url = $apiprotocol.$apiip.":".$apiport.$path;
```
Comment (JonasGroeger) [inline zones.php]: `$url = $apiprotocol . "://" . $apiip . ":" . $apiport . $path;`

File: zones.php [hunk 3]
```diff
@@ -11,14 +11,14 @@
...
 
     if ($authmethod == "auto") {
         $ad = curl_init();
         curl_setopt($ad, CURLOPT_HTTPHEADER, array('X-API-Key: '.$apipass));
-        curl_setopt($ad, CURLOPT_URL, "http://$apiip:$apiport/servers/localhost/statistics");
+        curl_setopt($ad, CURLOPT_URL, $apiprotocol.$apiip.":".$apiport."/servers/localhost/statistics");
```
Comment (JonasGroeger) [inline zones.php]: `curl_setopt($ad, CURLOPT_URL, $apiprotocol . "://" . $apiip . ":" . $apiport . "/servers/localhost/statistics");`

**Comment by extremeshok** [comment]
Please could this be merged or support for more (5) ns records be added.

**Comment by tuxis-ie** [comment]
It took me a while, sorry about that. This is now merged. I also included a thingy in misc.php to convert the old method to the new one.

Thanks a lot for your work. This is a well-requested feature!

**Comment by bfbones** [comment]
Thanks for merging!

**Pull request merged**